test/train-sets/ref/help.stdout

```diff
@@ -524,274 +228,8 @@ Gradient Descent Options:
-  --batch_sz_no_doubling      Batch_sz does not double
-Stochastic Variance Reduced Gradient Options:
-  --svrg                  Streaming Stochastic Variance Reduced Gradient
-  --stage_size arg (=1, ) Number of passes per SVRG stage
-Top K Options:
-  --top arg             Top k recommendation
-Make Multiclass into Warm-starting Contextual Bandit Options:
-  --warm_cb arg                        Convert multiclass on <k> classes into a
-                                       contextual bandit problem
-  --warm_cb_cs                         Consume cost-sensitive classification 
-                                       examples instead of multiclass
-  --loss0 arg (=0, )                   Loss for correct label
-  --loss1 arg (=1, )                   Loss for incorrect label
-  --warm_start arg (=0, )              Number of training examples for warm 
-                                       start phase
-  --epsilon arg                        Epsilon-greedy exploration
-  --interaction arg (=4294967295, )    Number of examples for the interactive 
-                                       contextual bandit learning phase
-  --warm_start_update                  Indicator of warm start updates
-  --interaction_update                 Indicator of interaction updates
-  --corrupt_type_warm_start arg (=1, ) Type of label corruption in the warm 
-                                       start phase (1: uniformly at random, 2: 
-                                       circular, 3: replacing with overwriting 
-                                       label). Choices: {1, 2, 3}
-  --corrupt_prob_warm_start arg (=0, ) Probability of label corruption in the 
-                                       warm start phase
-  --choices_lambda arg (=1, )          The number of candidate lambdas to 
-                                       aggregate (lambda is the importance 
-                                       weight parameter between the two 
-                                       sources)
-  --lambda_scheme arg (=1, )           The scheme for generating candidate 
-                                       lambda set (1: center lambda=0.5, 2: 
-                                       center lambda=0.5, min lambda=0, max 
-                                       lambda=1, 3: center lambda=epsilon/(1+ep
-                                       silon), 4: center lambda=epsilon/(1+epsi
-                                       lon), min lambda=0, max lambda=1); the 
-                                       rest of candidate lambda values are 
-                                       generated using a doubling scheme. 
-                                       Choices: {1, 2, 3, 4}
-  --overwrite_label arg (=1, )         The label used by type 3 corruptions 
-                                       (overwriting)
-  --sim_bandit                         Simulate contextual bandit updates on 
-                                       warm start examples
```

Comment:
There's a check in the help string generation function which does enabled reduction filtering. The check means that you will filter if there are **any** other options passed. Seems like it could cause confusion but I will work around it for this PR